src/App.vue

```diff
@@ -483,22 +485,38 @@ export default {
         console.error('WARNING: Legal Type in Legal db does not match Corp Type in Auth db!')
       }
 
+      // FUTURE: change this to a single setter/object?
+      this.setAdminFreeze(business.adminFreeze || false) // *** TODO: remove after API update
       this.setEntityName(business.legalName)
+      this.setEntityState(business.state || EntityState.ACTIVE) // *** TODO: remove after API update
       this.setEntityType(business.legalType)
-      this.setEntityStatus(business.goodStanding ? EntityStatus.GOOD_STANDING : EntityStatus.NOT_IN_COMPLIANCE)
-      this.setEntityBusinessNo(business.taxId) // may be empty
-      this.setEntityIncNo(business.identifier)
+      this.setBusinessNumber(business.taxId || null) // may be empty
+      this.setIdentifier(business.identifier)
       this.setEntityFoundingDate(this.apiToDate(business.foundingDate))
       this.setLastAnnualReportDate(business.lastAnnualReportDate) // may be empty
       this.setLastAddressChangeDate(business.lastAddressChangeDate) // may be empty
       this.setLastDirectorChangeDate(business.lastDirectorChangeDate) // may be empty
-
-      // *** TODO: get entity status (eg, Dissolved) -- see also business.goodStanding
-
-      // *** TODO: remove debugging code
-      business.dissolutionDate = this.dateToApi(new Date())
-      this.setEntityDissolutionDate(this.apiToDate(business.dissolutionDate))
-      this.setEntityDissolutionType(DissolutionTypes.HDO) // *** TODO: get from business object
+      this.setComplianceWarnings(Array.isArray(business.complianceWarnings) ? business.complianceWarnings : [])
+      this.setGoodStanding(business.goodStanding)
+
+      if (business.state === EntityState.HISTORICAL && business.stateFiling) {
+        // fetch the filing that changed the business state
+        const stateFiling = await this.fetchFiling(business.stateFiling)
+
+        // parse it (specific to dissolution filing atm)
+        const dissolutionType = stateFiling?.dissolution?.dissolutionType as DissolutionTypes
+        const effectiveDate = stateFiling?.header?.effectiveDate as string
+
+        // create the reason text to display in the info header
+        if (dissolutionType && effectiveDate) {
+          const name = this.dissolutionTypeToName(dissolutionType)
+          const enDash = '–' // ALT + 0150
```

Comment:
Note this self-documenting code to get an en dash (which is wider than a hyphen but not as wide as an em dash). Example:
first-second
first–second
first—second